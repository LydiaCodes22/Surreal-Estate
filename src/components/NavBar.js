import React from "react";
import "../styles/NavBar.css";
import { Link } from "react-router-dom";

const logo = require("../logo.png");

const NavBar = () => {
  return (
    <div className="navbar">
      <Link to="/">
        <img className="logo-image" src={logo} alt="Surreal Estate Logo" />
      </Link>
      <ul className="navbar-list">
        <li className="navbar-item">
          <Link to="/view-properties">View Properties</Link>
        </li>
        <li className="navbar-item">
          <Link to="/add-properties">Add a Property</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
