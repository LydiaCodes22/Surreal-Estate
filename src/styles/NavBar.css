.navbar {
    border-bottom: 10px solid rgb(231, 225, 225);
    display: flex;
    flex-wrap: wrap;
    align-items: center;    
}

.logo-image {
    width: 200px;
    min-width: 15%
}

.navbar-list {
    margin: 0;
    padding: 0;
}

.navbar-item {
    display: inline-block;
    margin: 0;
    padding: 10px;
}