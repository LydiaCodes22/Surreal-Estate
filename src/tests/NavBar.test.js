import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
// import userEvent from "@testing-library/user-event";
// import { createMemoryHistory } from "history";
import NavBar from "../components/NavBar";

describe("the NavBar", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <Router>
        <NavBar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the elements correctly", () => {
    render(
      <Router>
        <NavBar />
      </Router>
    );
    const viewPropertiesElement = screen.getByText(/view properties/i);
    const addPropertiesElement = screen.getByText(/add a property/i);
    const logoImage = screen.getByAltText(/Surreal Estate Logo/i);
    expect(viewPropertiesElement).toBeInTheDocument();
    expect(addPropertiesElement).toBeInTheDocument();
    expect(logoImage).toBeInTheDocument();
  });

  // it("has links to the right parts of the app", () => {
  //   const history = createMemoryHistory();
  //   render(
  //     <Router history={history}>
  //       <NavBar />
  //     </Router>
  //   );
  //   expect(history.location.pathname).toEqual("/");
  //   userEvent.click(screen.getByText(/view properties/i));
  //   expect(history.location.pathname).toEqual("/view-properties");
  //   userEvent.click(screen.getByText(/add a property/i));
  //   expect(history.location.pathname).toEqual("/add-properties");
  //   userEvent.click(screen.getByAltText(/Surreal Estate Logo/i));
  //   expect(history.location.pathname).toEqual("/");
  // });
});
