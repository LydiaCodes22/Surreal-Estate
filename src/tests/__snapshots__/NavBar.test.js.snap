// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the NavBar matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <a
      href="/"
    >
      <img
        alt="Surreal Estate Logo"
        class="logo-image"
        src="logo.png"
      />
    </a>
    <ul
      class="navbar-list"
    >
      <li
        class="navbar-item"
      >
        <a
          href="/view-properties"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          href="/add-properties"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
